scrunch/datasets.py

```diff
@@ -1088,6 +1044,47 @@ def editor(self, _):
             'Unsupported operation on the editor property'
         )
 
+    def change_editor(self, user):
+        """
+        Change the current editor of the Crunch dataset.
+
+        :param user:
+            The email address, the crunch url or a User instance of the user
+            who should be set as the new current editor of the given dataset.
+
+        :returns: None
+        """
+        def _host_from_url(url):
+            resolved = urlsplit(url)
+            return resolved.hostname
+
+        def _to_url(email):
+            api_users = 'https://{}/api/users/'.format(
```

Comment:
You don't need to construct the URL like this. It looks like the bigger problem here is how to obtain the site's root when you only have the catalog resource. For now, make a module level function like this:
```
def root_from_resource(resource):
    return resource.session.get('https://{}/api/'.format(urlsplit(resource.self)))
```